### PR TITLE
Add documentation for "QA Mode"

### PR DESCRIPTION
**Pull request by meretp**
### Summary of changes
 
Add some basic documentation about the QA mode as the described feature is restricted to this mode.

### Context and reason for change

On this branch we implement a new QA mode. The user should only be able to mark attributions as preferred if OpossumUI is in QA mode.
The rendered changes can be seen [here](https://github.com/meretp/OpossumUI/blob/document-restriction-for-preferred-attributions/USER_GUIDE.md#preferred-attributions-).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -375,3 +375,12 @@ In the audit view, an attributions can be marked as preferred, to indicate that
 Only signals with a source marked as `isRelevantForPreference` can be preferred over. If no signal source has this flag set, then the feature is disabled.
 
 To mark an attribution as preferred, choose an attribution in the audit view, and open the `...` context menu. From there, you can mark or unmark an attribution as preferred. When an attribution is marked as preferred, `preferred = true` is written to the .opossum file, and the origin IDs of all visible signals relevant for preference are written in the field `preferredOverOriginIds`. Preferred attributions are displayed with a star icon.
+
+Note that you are only able to mark an attribution as preferred (or unmark if it was preferred beforehand) if you are in
+"QA Mode". To enable this mode click the item "QA Mode" in the `View` submenu.
+
+![Screenshot from 2023-10-05 08-29-46.png](./docs/user_guide_screenshots/disabled_qa_mode.png)
```
Comment (benedikt-richter) [inline USER_GUIDE.md:382]: I'd remove the string `Screenshot from 2023-10-05 08-29-46.png`

File: USER_GUIDE.md [hunk 2]
```diff
@@ -375,3 +375,12 @@ In the audit view, an attributions can be marked as preferred, to indicate that
...
+
+If "QA Mode" is enabled the icon will change as in the screenshot below.
+
+![Screenshot from 2023-10-05 08-29-51.png](./docs/user_guide_screenshots/enabled_qa_mode.png)
```
Comment (benedikt-richter) [inline USER_GUIDE.md:386]: as above

**Review by benedikt-richter — APPROVED** [review]
lgtm 

just minor points.

For the future, we could also consider to document the QA mode on a higher level (currently, it is part of preferred documentation).  However, as it only affectes the preferred feature so far, that is fine with me.

**Comment by meretp** [comment]
Yes, I also thought about it as it reads like some kind of "higher-level" mode but as it only affects this feature, I decided to put it there for now.

**Pull request merged**